app/models/mixins/retirement_mixin.rb

```diff
@@ -10,9 +10,9 @@ module RetirementMixin
   end
 
   module ClassMethods
-    def make_retire_request
-      options = {:task => task, :userid => current_user, :ids => objs, :src_ids => objs}
-      (klass.to_s + "RetireRequest").constantize.make_request(@request_id, options, current_user)
+    def make_retire_request(*src_ids)
+      options = {:src_ids => src_ids.present? ? src_ids.first : id}
```

Comment:
Can you add some tests?  Since the array is not splatted above, I think you'll find that you end up with something like 
```ruby
{:src_ids => [[1,2,3]]}
```